### ISSUE TITLE
Exception on adding Route Guard

### ISSUE THREAD

**Issue by Im-PJ** [nylo-core/support#27]
I'm getting exception while adding route guard to a route as below.  

router.route(DashboardPage.path, (_) => DashboardPage()).addRouteGuard(AuthRouteGuard());

```
E/flutter ( 8167): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Unsupported operation: Cannot add to an unmodifiable list
E/flutter ( 8167): #0      UnmodifiableListMixin.add (dart:_internal/list.dart:114:5)
E/flutter ( 8167): #1      NyRouterRoute.addRouteGuard (package:nylo_support/router/models/nyrouter_route.dart:88:18)
E/flutter ( 8167): #2      appRouter.<anonymous closure> (package:nalco_netra/routes/router.dart:15:72)
E/flutter ( 8167): #3      nyRoutes (package:nylo_support/router/router.dart:30:8)
E/flutter ( 8167): #4      appRouter (package:nalco_netra/routes/router.dart:8:16)
E/flutter ( 8167): #5      RouteProvider.boot (package:nalco_netra/app/providers/route_provider.dart:7:20)
E/flutter ( 8167): #6      bootApplication (package:nylo_support/providers/providers.dart:17:39)
```

**Comment by agordn52** [comment]
Hi @Im-PJ,

Thanks for reporting this, I'll take a look now.

**Comment by agordn52** [comment]
This is resolved in [5.16.0](https://pub.dev/packages/nylo_framework). Thanks :)
